frontend/device/gesturedetector.lua

```diff
@@ -162,18 +165,24 @@ end
 --[[--
 Compares `last_pan` with `first_tev` in this slot.
 
+The second boolean argument `simple` results in only four directions if true.
+
 @return (direction, distance) pan direction and distance
 --]]
-function GestureDetector:getPath(slot)
-    local x_diff = self.last_tevs[slot].x - self.first_tevs[slot].x
-    local y_diff = self.last_tevs[slot].y - self.first_tevs[slot].y
+function GestureDetector:getPath(slot, simple, first_tev)
+    first_tev = first_tev or self.first_tevs
+
+    local x_diff = self.last_tevs[slot].x - first_tev[slot].x
+    local y_diff = self.last_tevs[slot].y - first_tev[slot].y
     local direction = nil
     local distance = math.sqrt(x_diff*x_diff + y_diff*y_diff)
     if x_diff ~= 0 or y_diff ~= 0 then
         local v_direction = y_diff < 0 and "north" or "south"
         local h_direction = x_diff < 0 and "west" or "east"
-        if math.abs(y_diff) > 0.577*math.abs(x_diff)
-            and math.abs(y_diff) < 1.732*math.abs(x_diff) then
+        if not simple
+           and math.abs(y_diff) > 0.577*math.abs(x_diff)
+           and math.abs(y_diff) < 1.732*math.abs(x_diff)
```

Comment:
Not your numbers I guess, but no comment says where 0.577 and 1.732 come from. Any idea?

Reply:
Looks like it's from the early days of the program in https://github.com/koreader/koreader/commit/15eccb4da7aa1afa44ffc0f6b8cbbb04abb521b8 but that does not, of course, answer the question.

Reply:
Also see [here](https://www.wolframalpha.com/input/?i=Euler%27s+constant&assumption=%7B%22C%22,+%22Euler%27s+constant%22%7D+-%3E+%7B%22NamedConstant%22,+%22dflt%22%7D&assumption=%7B%22ConstClash%22,+%22Euler%27s+constant%22%7D+-%3E+%7BE%7D) for the first. I knew it looked familiar.

The other one seems to be the result of √3.